### PR TITLE
Clean up recipe and re-enable linux tests

### LINKED ISSUE #16: Re-enable tests on Linux
The tests appear to block uploads from happening on Linux even though they don't fail. See issue ( https://github.com/conda-forge/erlang-feedstock/issues/1 ) and xrefs for details. As a result, we have disabled testing on Linux. It would be nice to figure out why this is and re-enable testing afterwards.

### PR DESCRIPTION
**Pull request by Anthchirp**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Resolves #16

### CODE REVIEW

**Comment by Anthchirp** [comment]
@conda-forge/erlang This is ready for review/merging

This includes the update to 23.1.2 and also re-enables the linux tests (#16).

Tests were skipped because of a too long shebang, but in this particular case we can fix that by just using normal path lookup mechanisms to find `perl`. The modified file also doesn't make it into the release.

I've also added myself as maintainer if that's okay with you. For that we need another round of @conda-forge-admin please rerender

**Comment by Anthchirp** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh [hunk 1]
```diff
@@ -11,22 +11,13 @@ export ERL_TOP="$(pwd)"
...
-if [ "$(uname)" == "Darwin" ]
-then
-    make release_tests
-    cd "${ERL_TOP}/release/tests/test_server"
-    ${ERL_TOP}/bin/erl -s ts install -s ts smoke_test batch -s init stop
-    cd ${ERL_TOP}
-fi
+# Fix up too long shebang line which is blocking tests on Linux
+# cf. https://github.com/conda-forge/erlang-feedstock/issues/16
+sed -i.bak -e '1 s@.*@#!/usr/bin/env perl@' make/make_emakefile
```
Comment (jakirkham) [inline recipe/build.sh:16]: Nice! Thanks for figuring this out 😄

**Comment by jakirkham** [comment]
Thanks Markus! 😄

**Pull request merged**